p2p/rlpx.go

```diff
@@ -160,7 +160,12 @@ func readProtocolHandshake(rw MsgReader, peer discover.NodeID) (*protoHandshake,
 	}
 	if msg.Code != handshakeMsg {
 		var emptyMsgObj []interface{}
-		log.Proto("<<UNEXPECTED_"+devp2pCodeToString[msg.Code], "obj", emptyMsgObj, "size", int(msg.Size), "peer", peer)
+		l := len(devp2pCodeToString)
+		if int(msg.Code) < l {
```

Comment:
It's probably better to do this check based on whether the key `msg.Code` is in `devp2pCodeToSTring` since it's actually a map and not a slice/array. It is only by coincidence that the keys are 0-indexed and consecutive

Reply:
that's what i thought too but apparently it is an array.
`if msgType, ok devp2pCodeToString[msg.Code]; ok {}`
this is what i tried first, but it shows error. doesn't return k, v. just the value.

Reply:
Oh good call. In that case, it'd be better IMO to just do `if int(msg.Code) < len(devp2pCodeToString) {`

Reply:
got it